### ISSUE TITLE
compile: .... methods should return CompiledMethod object

### ISSUE THREAD

**Issue by demarey** [guillep/pharo-core#12880]
Strangely, all compile: .... methods return a selector.
They should return the `CompiledMethod` object.

**Comment by privat** [comment]
Do you have an example?

```st
(Smalltalk compiler compile: 'foo ^1') class >>> CompiledMethod 
```

**Comment by demarey** [comment]
```smalltalk
ClassDescription>>#compile:classified:notifying:
String compile: 'foo' classified: 'foo' notifying: nil >>> "#foo"
```

**Comment by privat** [comment]
My guess it's that a legacy thing and since that there are a lot of clients that expect that such methods return selectors there is no real gain to change that. Maybe @MarcusDenker knows the history better.

**Comment by MarcusDenker** [comment]
yes, this is legacy.. I wonder if it is not better to add new APIs and keep the old ones compatible

**Comment by privat** [comment]
I opened a meta-issue #12883 to rant about the compilation API for instance :)

**Comment by privat** [comment]
My position on this specific issue: `compile:*` is legacy, maybe we could deprecate it? The way forward is the `compiler` message. But the API is still under heavy changes. It could be a Pahro12 goal

```st
String compile: 'foo' classified: 'foo' notifying: nil >>> "#foo"
```

should become something like

```st
String compiler protocol: 'foo'; install: 'foo' >>> aCompiledMethod
```
